### PR TITLE
Update Developer Guide for UI Design and List Toggling Functionality

### LINKED ISSUE #47: Update Developer Guide
On Architecture Diagrams and EditCommands implementations.

### PR DESCRIPTION
**Pull request by weixue123**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -19,6 +19,28 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+The `UI` component:
+* Executes users' commands using the `Logic` component.
+* Listens for changes to `Model` data so that the UI can be updated with the modified data. 
+* Any changes in `Model` data, i.e. customers, cheeses or orders data, are reflected through the `Panels` and `Cards`
+  sub-components.
+  
+The class diagram below shows in more detail the compositions of the `Panels` and `Cards` components as well as their 
+relationships with other classes.
+
+![Structure of the Panels & Cards Component](images/UiPanelsCardsClassDiagram.png)
```
Comment (daniellau88) [inline docs/DeveloperGuide.md:43]: Hi, may I know the link for the diagram

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -27,8 +49,43 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 ### Common classes
 
+--------------------------------------------------------------------------------------------------------------------
+
 ## **Implementation** [In Progress]
```
Comment (daniellau88) [inline docs/DeveloperGuide.md]: Also can consider removing the `[In Progress]`

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (weixue123) [reply docs/DeveloperGuide.md:43]: I did it using plant UML leh (same as in AB3) so you will have to pull the branch to see it. I will paste them here so it's more convenient to review.

UiClassDiagram

UiPanelsCardsClassDiagram

ListPanelTogglingSequenceDiagram

Comment (weixue123) [reply docs/DeveloperGuide.md:43]: I realized how ugly the panels and cards diagram is - gimme a min ill make some edits.

Comment (weixue123) [reply docs/DeveloperGuide.md:43]: UiPanelsCardsClassDiagram

Comment (daniellau88) [reply docs/DeveloperGuide.md:43]: This looks gd

Comment (daniellau88) [reply docs/DeveloperGuide.md:43]: Should the sequence diagram contain the strings? it seems like a bit out of place

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -19,6 +19,29 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
 ### UI component
 
+![Structure of the UI Component](images/UiClassDiagram.png)
+
+**API**: `Ui.java`
+
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `StatusBarFooter` etc. All
+these, including the `MainWindow`, inherit from the abstract `UiPart` class.
+
+The `UI` component uses the JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that
```
Comment (daniellau88) [inline docs/DeveloperGuide.md]: Mini issue: JavaFx is JavaFX

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -19,6 +19,29 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+
+* Executes users' commands using the `Logic` component.
+* Listens for changes to `Model` data so that the UI can be updated with the modified data.
+* Any changes in `Model` data, i.e. customers, cheeses or orders data, are reflected through the `Panels` and `Cards`
+  sub-components.
+
+The class diagram below shows in more detail the compositions of the `Panels` and `Cards` components as well as their
+relationships with other classes.
+
+![Structure of the Panels & Cards Component](images/UiPanelsCardsClassDiagram.png)
```
Comment (daniellau88) [inline docs/DeveloperGuide.md:43]: Should the sequence diagram contain the strings? it seems like a bit out of place

Comment (daniellau88) [inline docs/DeveloperGuide.md:43]: Also the `CHEESE_LIST` what is it suppose to represent?

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (weixue123) [reply docs/DeveloperGuide.md]: okay noted

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (weixue123) [reply docs/DeveloperGuide.md:43]: It's supposed to be the enum representing which list to show. Actually we may not need to spell it out in the sequence diagram.

Comment (daniellau88) [reply docs/DeveloperGuide.md:43]: Sure, I just pointed out because there seems to be no explanation for it haha

Comment (weixue123) [reply docs/DeveloperGuide.md:43]: The updated sequence diagram.

ListPanelTogglingSequenceDiagram

I think we can leave the "listcheeses" argument in? I referred to AB3's developer guide and this seems like the style they are following

**Review by daniellau88 — APPROVED** [review]
LGTM

**Pull request merged**